packages/esbuild-plugin/src/index.ts

```diff
@@ -0,0 +1,133 @@
+import { Util } from '@honeybadger-io/core'
+import { cleanOptions, sendDeployNotification, uploadSourcemaps, Types } from '@honeybadger-io/plugin-core'
+import type { Message, OutputFile, PluginBuild } from 'esbuild'
+import { promises } from 'fs'
+import { dirname, basename, join } from 'path'
+
+const writeFile = promises.writeFile
+const mkdir = promises.mkdir
+
+const PLUGIN_NAME = 'HoneybadgerSourceMapPlugin'
+
+class HoneybadgerSourceMapPlugin {
+
+  private static idSeed = 0
+
+  private readonly options: Types.HbPluginOptions
+
+  constructor(options: Types.HbPluginUserOptions) {
+    this.options = cleanOptions(options)
+  }
+  setup() {
+    const self = this;
+
+    return {
+      name: PLUGIN_NAME,
+      setup(build: PluginBuild) {
+        if (self.isNonProdEnv()) {
+          return
+        }
+
+        // need to set write to false to get the outputFiles
+        build.initialOptions.write = false
```

Comment:
Apparently we have to [disable writing to disk](https://github.com/evanw/esbuild/issues/3408) in order to get the output files, so we do that ourselves below.